lib/astutils.cpp

```diff
@@ -3293,7 +3293,7 @@ static ExprUsage getFunctionUsage(const Token* tok, int indirect, const Settings
     } else if (ftok->str() == "{") {
         return indirect == 0 ? ExprUsage::Used : ExprUsage::Inconclusive;
     } else if (ftok->variable() && ftok == ftok->variable()->nameToken()) { // variable init/constructor call
-        if (ftok->variable()->type() && ftok->variable()->type()->needInitialization == Type::NeedInitialization::True)
+        if (ftok->variable()->type() && ftok->variable()->type()->classScope && ftok->variable()->type()->classScope->numConstructors == 0)
```

Comment:
Will we still diagnose this case?

```cpp
struct S {
    S(char*);
    int i;
};
void f() {
    char c;
    S s(&c);
}
```

Reply:
No. But if we did warn, it was by accident, since there is no argument checking at all.
That line was originally added in https://github.com/danmar/cppcheck/pull/5713/, which is about POD types.